service-proxy.js

```diff
@@ -494,6 +492,68 @@ function deletePeerIndex(serviceName, hostPort) {
     deleteIndexEntry(self.knownPeers, hostPort, serviceName);
 };
 
+ServiceDispatchHandler.prototype.ensurePeerConnected =
+function ensurePeerConnected(peer, reason) {
+    var self = this;
+
+    if (peer.isConnected('out')) {
+        return;
+    }
+
+    peer.waitForIdentified(peer.connectTo(), onConnIded);
+
+    function onConnIded(err) {
+        if (err) {
+            self.logger.warn(
+                'failed to ensure outgoing connection to service peer',
+                self.extendLogInfo({
+                    error: err,
+                    peerHostPort: peer.hostPort,
+                    refreshReason: reason
+                }));
+            return;
+        }
+        peer.drain({
+            reason: reason,
```

Comment:
We either need to communicate to every one to expect declined messages about draining.

Or make a change in the hyperbahn client in all languages to silence the draining error frame log.